src/solvers/base_solver.cpp

```diff
@@ -55,57 +54,56 @@ void BaseSolver::SetNaturalBCs(mfem::Array<int> &nat_bdr, mfem::Coefficient *nat
   m_nat_bdr_coef = nat_bdr_coef;
 }
 
-void BaseSolver::SetState(const mfem::Array<FiniteElementState> &state)
+void BaseSolver::SetState(const std::map<std::string, FiniteElementState> &state)
```

Comment:
Can you explain why you changed from an array to an ordered map?

Reply:
related to https://github.com/LLNL/serac/issues/23

Also in an early test, the mfem::Array was segfaulting on deconstruction when it contained std::shared_ptrs while std containers weren't.

Reply:
As stated in #23, perhaps std::unordered_map would be another option as there isn't any explicit ordering with the state mapping.